arraylist/src/main/java/ua/hillel/ArrayList.java

```diff
@@ -1,68 +1,138 @@
 package ua.hillel;
 
+import java.util.Arrays;
+import java.util.Objects;
+
 /**
  * Implement a list that is based on array. This resizable data structure
  * is based on an array and is a simplified version of {@link java.util.ArrayList}.
  */
 public class ArrayList<T> implements List<T> {
 
+    transient Object[] elementData;
+
+    private static final int DEFAULT_CAPACITY = 5;
+    public static Object[] emptyArray = {};
+
+    public ArrayList() {
+        this.elementData = emptyArray;
+    }
+
+    private int size;
+
     @Override
     public void add(T element) {
-
+        Objects.requireNonNull(element);
+        int indexElement = size;
+        if (size == emptyArray.length) {
+            elementData = new Object[DEFAULT_CAPACITY];
+        }
+        if (size == elementData.length) {
+            elementData = Arrays.copyOf(elementData, size + 1);
+        }
+        elementData[indexElement] = element;
+        size = size + 1;
     }
 
     @Override
     public void add(int index, T element) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Such index does not exist in this array does not exist");
+        }
+        Objects.requireNonNull(element);
+        if (size == elementData.length) {
+            elementData = Arrays.copyOf(elementData, size + 1);
+        }
+        System.arraycopy(this.elementData, index, this.elementData, index + 1, size - index);
+        this.elementData[index] = element;
+        size = size + 1;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("The requested index is not in the range of the array");
+        }
+        if (elementData[index] == (null)) {
+            throw new NullPointerException();
+        }
+        return (T) elementData[index];
     }
 
     @Override
     public T get(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (elementData[i].equals(element)) {
+                return element;
+            }
+        }
+        throw new IllegalArgumentException("Such element was not found in the list");
     }
 
     @Override
     public T getFirst() {
-        return null;
+        if (elementData[0] == null) {
+            throw new NullPointerException("The first index of the list is null.");
+        }
+        return get(0);
     }
 
     @Override
     public T getLast() {
-        return null;
+        if (elementData[size - 1] == null) {
+            throw new NullPointerException("The last index of the list is null.");
+        }
+        return get(size - 1);
     }
 
     @Override
     public void set(int index, T element) {
-
+        if (elementData[index] == null) {
+            throw new NullPointerException("The requested index is null");
+        }
```

Comment:
What are you checking here? An element can be null, it is not a problem. Moreover, You're not check-in an index here, you're checking an element, which is wrong. You can check the index as you did in `public T get(int index)`, for example.